<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>RNAlib-2.4.3: SWIG Wrapper Notes</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">RNAlib-2.4.3
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('wrappers.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">SWIG Wrapper Notes </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>Below, you find an enumeration of special note on how functions, structures, enums, and macro definitions are wrapped.</p>
<dl class="reflist">
<dt><a class="anchor" id="_wrappers000002"></a>Global <a class="el" href="group__aln__utils.html#gaa12b481a7e7b965ef2eb1bcc4399e759">vrna_aln_conservation_col</a>  (const char **alignment, const vrna_md_t *md_p, unsigned int options)</dt>
<dd>This function is available in an overloaded form where the last two parameters may be omitted, indicating <code>md</code> = <em>NULL</em>, and <code>options</code> = <a class="el" href="group__aln__utils.html#ga1e659227c9fc077d29989f576f129000" title="Flag indicating Shannon Entropy measure. ">VRNA_MEASURE_SHANNON_ENTROPY</a>, respectively.  </dd>
<dt><a class="anchor" id="_wrappers000001"></a>Global <a class="el" href="group__aln__utils.html#gab6f16a2ea93f3bfd4d089cc8d448bb16">vrna_aln_conservation_struct</a>  (const char **alignment, const char *structure, const vrna_md_t *md)</dt>
<dd>This function is available in an overloaded form where the last parameter may be omitted, indicating <code>md</code> = <em>NULL</em>  </dd>
<dt><a class="anchor" id="_wrappers000005"></a>Global <a class="el" href="group__file__utils.html#ga5e993fc4b9602af73aaaab4d3b3cd9a9">vrna_commands_apply</a>  (vrna_fold_compound_t *vc, vrna_cmd_t *commands, unsigned int options)</dt>
<dd>This function is attached as method <b>commands_apply()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000059"></a>Global <a class="el" href="group__struct__utils.html#gae966b9f44168a4f4b39ca42ffb5f37b7">vrna_db_flatten</a>  (char *structure, unsigned int options)</dt>
<dd><p class="startdd">This function flattens an input structure string in-place! The second parameter is optional and defaults to <a class="el" href="group__struct__utils.html#ga559ebf76b1b289f85309f4206e99aa1a" title="Default bitmask to indicate secondary structure notation using any pair of brackets. ">VRNA_BRACKETS_DEFAULT</a>.</p>
<p class="enddd">An overloaded version of this function exists, where an additional second parameter can be passed to specify the target brackets, i.e. the type of matching pair characters all brackets will be flattened to. Therefore, in the scripting language interface this function is a replacement for <a class="el" href="group__struct__utils.html#ga690425199c8b71545e7196e3af1436f8" title="Substitute pairs of brackets in a string with another type of pair characters. ">vrna_db_flatten_to()</a>.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000060"></a>Global <a class="el" href="group__struct__utils.html#ga690425199c8b71545e7196e3af1436f8">vrna_db_flatten_to</a>  (char *string, const char target[3], unsigned int options)</dt>
<dd>This function is available as an overloaded version of <a class="el" href="group__struct__utils.html#gae966b9f44168a4f4b39ca42ffb5f37b7" title="Substitute pairs of brackets in a string with parenthesis. ">vrna_db_flatten()</a>  </dd>
<dt><a class="anchor" id="_wrappers000003"></a>Global <a class="el" href="group__utils.html#gae081ac655a76bd5c4b3d86c60b096b75">vrna_enumerate_necklaces</a>  (const unsigned int *type_counts)</dt>
<dd>This function is available as global function <b>enumerate_necklaces()</b> which accepts lists input, an produces list of lists output.  </dd>
<dt><a class="anchor" id="_wrappers000029"></a>Global <a class="el" href="group__eval.html#ga6cea75c0eb9857fb59172be54cab09e0">vrna_eval_covar_structure</a>  (vrna_fold_compound_t *vc, const char *structure)</dt>
<dd>This function is attached as method <b>eval_covar_structure()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000033"></a>Global <a class="el" href="group__loops.html#gab3eb4651dc26dc2b653a57dd340d7e68">vrna_eval_hp_loop</a>  (vrna_fold_compound_t *vc, int i, int j)</dt>
<dd>This function is attached as method <b>eval_hp_loop()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000034"></a>Global <a class="el" href="group__loops.html#ga018f1314dbbae42fdd27c94670b61721">vrna_eval_int_loop</a>  (vrna_fold_compound_t *vc, int i, int j, int k, int l)</dt>
<dd>This function is attached as method <b>eval_int_loop()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000030"></a>Global <a class="el" href="group__eval.html#ga730ba4df55c02fd530a0cddd49faf760">vrna_eval_loop_pt</a>  (vrna_fold_compound_t *vc, int i, const short *pt)</dt>
<dd>This function is attached as method <b>eval_loop_pt()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000031"></a>Global <a class="el" href="group__eval.html#gaff1b9e4f4d17b434b0a822fe783672c1">vrna_eval_move</a>  (vrna_fold_compound_t *vc, const char *structure, int m1, int m2)</dt>
<dd>This function is attached as method <b>eval_move()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000032"></a>Global <a class="el" href="group__eval.html#ga123dabc119ea98c968a5e903cc46f0fb">vrna_eval_move_pt</a>  (vrna_fold_compound_t *vc, short *pt, int m1, int m2)</dt>
<dd>This function is attached as method <b>eval_move_pt()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000025"></a>Global <a class="el" href="group__eval.html#ga58f199f1438d794a265f3b27fc8ea631">vrna_eval_structure</a>  (vrna_fold_compound_t *vc, const char *structure)</dt>
<dd>This function is attached as method <b>eval_structure()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000026"></a>Global <a class="el" href="group__eval.html#gadbd09372ddfd7a450bbd590c96a6bfe4">vrna_eval_structure_pt</a>  (vrna_fold_compound_t *vc, const short *pt)</dt>
<dd>This function is attached as method <b>eval_structure_pt()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000028"></a>Global <a class="el" href="group__eval.html#ga8a517cfeeae8c376ae7b1e0c401d38b4">vrna_eval_structure_pt_verbose</a>  (vrna_fold_compound_t *vc, const short *pt, FILE *file)</dt>
<dd>This function is attached as method <b>eval_structure_pt_verbose()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000027"></a>Global <a class="el" href="group__eval.html#ga0928d699d310178f84ee2351034e5cb5">vrna_eval_structure_verbose</a>  (vrna_fold_compound_t *vc, const char *structure, FILE *file)</dt>
<dd>This function is attached as method <b>eval_structure_verbose()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000051"></a>Global <a class="el" href="group__energy__parameters.html#gad607bc3a5b5da16400e2ca4ea5560233">vrna_exp_params_rescale</a>  (vrna_fold_compound_t *vc, double *mfe)</dt>
<dd><p class="startdd">This function is attached to <a class="el" href="group__fold__compound.html#structvrna__fc__s" title="The most basic data structure required by many functions throughout the RNAlib. ">vrna_fc_s</a> objects as overloaded <b>exp_params_rescale()</b> method.</p>
<p class="enddd">When no parameter is passed to this method, the resulting action is the same as passing <em>NULL</em> as second parameter to <a class="el" href="group__energy__parameters.html#gad607bc3a5b5da16400e2ca4ea5560233" title="Rescale Boltzmann factors for partition function computations. ">vrna_exp_params_rescale()</a>, i.e. default scaling of the partition function. Passing an energy in kcal/mol, e.g. as retrieved by a previous call to the <em>mfe()</em> method, instructs all subsequent calls to scale the partition function accordingly.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000053"></a>Global <a class="el" href="group__energy__parameters.html#gaa5409218068be84d7b50c78fbdaa85a9">vrna_exp_params_reset</a>  (vrna_fold_compound_t *vc, vrna_md_t *md_p)</dt>
<dd><p class="startdd">This function is attached to <a class="el" href="group__fold__compound.html#structvrna__fc__s" title="The most basic data structure required by many functions throughout the RNAlib. ">vrna_fc_s</a> objects as overloaded <b>exp_params_reset()</b> method.</p>
<p class="enddd">When no parameter is passed to this method, the resulting action is the same as passing <em>NULL</em> as second parameter to <a class="el" href="group__energy__parameters.html#gaa5409218068be84d7b50c78fbdaa85a9" title="Reset Boltzmann factors for partition function computations within a vrna_fold_compound_t according t...">vrna_exp_params_reset()</a>, i.e. global default model settings are used. Passing an object of type <a class="el" href="group__model__details.html#structvrna__md__s" title="The data structure that contains the complete model details used throughout the calculations. ">vrna_md_s</a> resets the fold compound according to the specifications stored within the <a class="el" href="group__model__details.html#structvrna__md__s" title="The data structure that contains the complete model details used throughout the calculations. ">vrna_md_s</a> object.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000039"></a>Class <a class="el" href="group__fold__compound.html">vrna_fc_s</a>  </dt>
<dd><p class="startdd"></p>
<p>This data structure is wrapped as an object <b>fold_compound</b> with several related functions attached as methods.</p>
<p class="enddd">A new <b>fold_compound</b> can be obtained by calling one of its constructors:<br />
</p><ul>
<li><em>fold_compound(seq)</em> &ndash; Initialize with a single sequence, or two concatenated sequences separated by an ampersand character '&amp;' (for cofolding)</li>
<li><em>fold_compound(aln)</em> &ndash; Initialize with a sequence alignment <em>aln</em> stored as a list of sequences (with gap characters)</li>
</ul>
<p>The resulting object has a list of attached methods which in most cases directly correspond to functions that mainly operate on the corresponding <em>C</em> data structure:<br />
</p><ul>
<li><em><a class="el" href="group__fold__compound.html#ac5eab693deac9a1a40c2a95ac294707c" title="The type of the vrna_fold_compound_t. ">type()</a></em> &ndash; Get the type of the <em>fold_compound</em> (See <a class="el" href="group__fold__compound.html#ga01a4ff86fa71deaaa5d1abbd95a1447d" title="An enumerator that is used to specify the type of a vrna_fold_compound_t. ">vrna_fc_type_e</a>)</li>
<li><em><a class="el" href="group__fold__compound.html#a95fbfed770b858e50c766505dc4bf998" title="The length of the sequence (or sequence alignment) ">length()</a></em> &ndash; Get the length of the sequence(s) or alignment stored within the <em>fold_compound</em> </li>
</ul>
</dd>
<dt><a class="anchor" id="_wrappers000004"></a>Global <a class="el" href="group__file__utils.html#gadbe8c9622f7bcc6dcbe3448b98df8656">vrna_file_commands_apply</a>  (vrna_fold_compound_t *vc, const char *filename, unsigned int options)</dt>
<dd>This function is attached as method <b>file_commands_apply()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000035"></a>Global <a class="el" href="group__file__utils.html#ga627ac281b5f11c63861726e6472626c9">vrna_file_msa_detect_format</a>  (const char *filename, unsigned int options)</dt>
<dd>This function exists as an overloaded version where the <code>options</code> parameter may be omitted! In that case, the <code>options</code> parameter defaults to <a class="el" href="group__file__utils.html#ga4acc255373831856a8417b68de8a94c0" title="Option flag indicating the set of default file formats. ">VRNA_FILE_FORMAT_MSA_DEFAULT</a>.  </dd>
<dt><a class="anchor" id="_wrappers000036"></a>Global <a class="el" href="group__file__utils.html#ga08a01c40ac5f5e0e04e9ae2258c99aa6">vrna_file_msa_read</a>  (const char *filename, char ***names, char ***aln, char **id, char **structure, unsigned int options)</dt>
<dd><p class="startdd">In the target scripting language, only the first and last argument, <code>filename</code> and <code>options</code>, are passed to the corresponding function. The other arguments, which serve as output in the C-library, are available as additional return values. Hence, a function call in python may look like this:</p>
<div class="fragment"><div class="line">num_seq, names, aln, id, structure = RNA.file_msa_read(<span class="stringliteral">&quot;msa.stk&quot;</span>, RNA.FILE_FORMAT_MSA_STOCKHOLM)</div></div><!-- fragment --><p>After successfully reading the first record, the variable <code>num_seq</code> contains the number of sequences in the alignment (the actual return value of the C-function), while the variables <code>names</code>, <code>aln</code>, <code>id</code>, and <code>structure</code> are lists of the sequence names and aligned sequences, as well as strings holding the alignment ID and the structure as stated in the <code>SS_cons</code> line, respectively. Note, the last two return values may be empty strings in case the alignment does not provide the required data.</p>
<p class="enddd">This function exists as an overloaded version where the <code>options</code> parameter may be omitted! In that case, the <code>options</code> parameter defaults to <a class="el" href="group__file__utils.html#ga62be992445cd8ab2ad7a8fded944338b" title="Option flag indicating Stockholm 1.0 formatted files. ">VRNA_FILE_FORMAT_MSA_STOCKHOLM</a>.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000037"></a>Global <a class="el" href="group__file__utils.html#ga59204cd1daa4927f5127cc65a2886efd">vrna_file_msa_read_record</a>  (FILE *fp, char ***names, char ***aln, char **id, char **structure, unsigned int options)</dt>
<dd><p class="startdd">In the target scripting language, only the first and last argument, <code>fp</code> and <code>options</code>, are passed to the corresponding function. The other arguments, which serve as output in the C-library, are available as additional return values. Hence, a function call in python may look like this:</p>
<div class="fragment"><div class="line">f = open(<span class="stringliteral">&#39;msa.stk&#39;</span>, <span class="charliteral">&#39;r&#39;</span>)</div><div class="line">num_seq, names, aln, <span class="keywordtype">id</span>, structure = RNA.file_msa_read_record(f, RNA.FILE_FORMAT_MSA_STOCKHOLM)</div><div class="line">f.close() </div></div><!-- fragment --><p>After successfully reading the first record, the variable <code>num_seq</code> contains the number of sequences in the alignment (the actual return value of the C-function), while the variables <code>names</code>, <code>aln</code>, <code>id</code>, and <code>structure</code> are lists of the sequence names and aligned sequences, as well as strings holding the alignment ID and the structure as stated in the <code>SS_cons</code> line, respectively. Note, the last two return values may be empty strings in case the alignment does not provide the required data.</p>
<p class="enddd">This function exists as an overloaded version where the <code>options</code> parameter may be omitted! In that case, the <code>options</code> parameter defaults to <a class="el" href="group__file__utils.html#ga62be992445cd8ab2ad7a8fded944338b" title="Option flag indicating Stockholm 1.0 formatted files. ">VRNA_FILE_FORMAT_MSA_STOCKHOLM</a>.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000038"></a>Global <a class="el" href="group__file__utils.html#ga0c90faf1515ce0b16620036b1055ff1e">vrna_file_msa_write</a>  (const char *filename, const char **names, const char **aln, const char *id, const char *structure, const char *source, unsigned int options)</dt>
<dd>In the target scripting language, this function exists as a set of overloaded versions, where the last four parameters may be omitted. If the <code>options</code> parameter is missing the options default to (<a class="el" href="group__file__utils.html#ga62be992445cd8ab2ad7a8fded944338b" title="Option flag indicating Stockholm 1.0 formatted files. ">VRNA_FILE_FORMAT_MSA_STOCKHOLM</a> | <a class="el" href="group__file__utils.html#ga1577ea0f497d9c501549c863a4f2c089" title="Option flag indicating to append data to a multiple sequence alignment file rather than overwriting i...">VRNA_FILE_FORMAT_MSA_APPEND</a>).  </dd>
<dt><a class="anchor" id="_wrappers000010"></a>Global <a class="el" href="group__hard__constraints.html#ga5b4de3247b67358080c176b94591a8e6">vrna_hc_add_from_db</a>  (vrna_fold_compound_t *vc, const char *constraint, unsigned int options)</dt>
<dd>This function is attached as method <b>hc_add_from_db()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000006"></a>Global <a class="el" href="group__hard__constraints.html#ga36ff456c43bf920629cee5a236e4f0ff">vrna_hc_init</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b>hc_init()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000049"></a>Class <a class="el" href="group__model__details.html">vrna_md_s</a>  </dt>
<dd><p class="startdd">This data structure is wrapped as an object <b>md</b> with multiple related functions attached as methods.</p>
<p>A new set of default parameters can be obtained by calling the constructure of <b>md:</b> <br />
</p><ul>
<li><em>md()</em> &ndash; Initialize with default settings</li>
</ul>
<p>The resulting object has a list of attached methods which directly correspond to functions that mainly operate on the corresponding <em>C</em> data structure:<br />
</p><ul>
<li><em>reset()</em> &ndash; <a class="el" href="group__model__details.html#ga8ac6ff84936282436f822644bf841f66">vrna_md_set_default()</a></li>
<li><em>set_from_globals()</em> &ndash; <a class="el" href="group__model__details.html#gabad896c3650d420f3f3ddefc69e2bceb">set_model_details()</a></li>
<li><em>option_string()</em> &ndash; <a class="el" href="group__model__details.html#ga3a7469f0725a849af6ba61a57dfd60ce">vrna_md_option_string()</a></li>
</ul>
<p>Note, that default parameters can be modified by directly setting any of the following global variables. Internally, getting/setting default parameters using their global variable representative translates into calls of the following functions, therefore these wrappers for these functions do not exist in the scripting language interface(s):</p>
<table class="doxtable">
<tr>
<th>global variable </th><th><em>C</em> getter </th><th><em>C</em> setter  </th></tr>
<tr>
<td>temperature </td><td><a class="el" href="group__model__details.html#ga96b24a74437f9ba46c4e06343155bf46" title="Get default temperature for energy evaluation of loops. ">vrna_md_defaults_temperature_get()</a> </td><td><a class="el" href="group__model__details.html#gaf9e527e9a2f7e6fd6e42bc6e602f5445" title="Set default temperature for energy evaluation of loops. ">vrna_md_defaults_temperature()</a> </td></tr>
<tr>
<td>dangles </td><td><a class="el" href="group__model__details.html#ga67ca06f95ae133778c79a4493c9817b8" title="Get default dangle model for structure prediction. ">vrna_md_defaults_dangles_get()</a> </td><td><a class="el" href="group__model__details.html#gac76a5374def8e5e4e644ff6e4cc72dee" title="Set default dangle model for structure prediction. ">vrna_md_defaults_dangles()</a> </td></tr>
<tr>
<td>betaScale </td><td><a class="el" href="group__model__details.html#gabb8780f5410c52f970d75b044059bd09" title="Get default scaling factor of thermodynamic temperature in Boltzmann factors. ">vrna_md_defaults_betaScale_get()</a> </td><td><a class="el" href="group__model__details.html#gae984567db36c3f9b8731ecc917abf3a2" title="Set default scaling factor of thermodynamic temperature in Boltzmann factors. ">vrna_md_defaults_betaScale()</a> </td></tr>
<tr>
<td>tetra_loop </td><td>this is an alias of <em>special_hp</em> </td><td></td></tr>
<tr>
<td>special_hp </td><td><a class="el" href="group__model__details.html#ga1d68a6efdaa1253cc63fd9cd06452559" title="Get default behavior for lookup of tabulated free energies for special hairpin loops, such as Tri-, Tetra-, or Hexa-loops. ">vrna_md_defaults_special_hp_get()</a> </td><td><a class="el" href="group__model__details.html#gafff6449a02744add0308e653230c15fc" title="Set default behavior for lookup of tabulated free energies for special hairpin loops, such as Tri-, Tetra-, or Hexa-loops. ">vrna_md_defaults_special_hp()</a> </td></tr>
<tr>
<td>noLonelyPairs </td><td>this is an alias of <em>noLP</em> </td><td></td></tr>
<tr>
<td>noLP </td><td><a class="el" href="group__model__details.html#ga934344888fbacaed538bbbfe910f2aa6" title="Get default behavior for prediction of canonical secondary structures. ">vrna_md_defaults_noLP_get()</a> </td><td><a class="el" href="group__model__details.html#ga2f88ffc393ac9d7987849c965fd29ea8" title="Set default behavior for prediction of canonical secondary structures. ">vrna_md_defaults_noLP()</a> </td></tr>
<tr>
<td>noGU </td><td><a class="el" href="group__model__details.html#ga5faa7d4e536d7fe36ec25428c0cf2563" title="Get default behavior for treatment of G-U wobble pairs. ">vrna_md_defaults_noGU_get()</a> </td><td><a class="el" href="group__model__details.html#ga98218f85c7a957a1d1ddf4627fdf5a39" title="Set default behavior for treatment of G-U wobble pairs. ">vrna_md_defaults_noGU()</a> </td></tr>
<tr>
<td>no_closingGU </td><td>this is an alias of <em>noGUclosure</em> </td><td></td></tr>
<tr>
<td>noGUclosure </td><td><a class="el" href="group__model__details.html#ga4f7fdad083243a5348d63320ddaa70f3" title="Get default behavior for G-U pairs as closing pair for loops. ">vrna_md_defaults_noGUclosure_get()</a> </td><td><a class="el" href="group__model__details.html#gade5b9951d71ca2fb357a4e6c0c18ccd1" title="Set default behavior for G-U pairs as closing pair for loops. ">vrna_md_defaults_noGUclosure()</a> </td></tr>
<tr>
<td>logML </td><td><a class="el" href="group__model__details.html#ga93f04e070d529c5d0bb87c9681f6ad29" title="Get default behavior recomputing free energies of multi-branch loops using a logarithmic model...">vrna_md_defaults_logML_get()</a> </td><td><a class="el" href="group__model__details.html#ga3de50a73455d88c3957386933b8e1f90" title="Set default behavior recomputing free energies of multi-branch loops using a logarithmic model...">vrna_md_defaults_logML()</a> </td></tr>
<tr>
<td>circ </td><td><a class="el" href="group__model__details.html#gad3a7e58de344ad93a08925f58f94f6fb" title="Get default behavior whether input sequences are circularized. ">vrna_md_defaults_circ_get()</a> </td><td><a class="el" href="group__model__details.html#ga4e1deb3e91a8a99e5c6dd905a5eb0186" title="Set default behavior whether input sequences are circularized. ">vrna_md_defaults_circ()</a> </td></tr>
<tr>
<td>gquad </td><td><a class="el" href="group__model__details.html#gae645b8612f879eb38b45244fa9eddb9e" title="Get default behavior for treatment of G-Quadruplexes. ">vrna_md_defaults_gquad_get()</a> </td><td><a class="el" href="group__model__details.html#ga0685ca2aeb39af76f2421fc308163dce" title="Set default behavior for treatment of G-Quadruplexes. ">vrna_md_defaults_gquad()</a> </td></tr>
<tr>
<td>uniq_ML </td><td><a class="el" href="group__model__details.html#gab48e70fd024bf838404bcbcca0c874a0" title="Get default behavior for creating additional matrix for unique multi-branch loop prediction. ">vrna_md_defaults_uniq_ML_get()</a> </td><td><a class="el" href="group__model__details.html#ga59b944f61c5d2babec2d4c48c820de67" title="Set default behavior for creating additional matrix for unique multi-branch loop prediction. ">vrna_md_defaults_uniq_ML()</a> </td></tr>
<tr>
<td>energy_set </td><td><a class="el" href="group__model__details.html#ga017ed6afb1cba2b7f242412cab618b53" title="Get default energy set. ">vrna_md_defaults_energy_set_get()</a> </td><td><a class="el" href="group__model__details.html#ga8dd29c55787a4576277e1907e92d810c" title="Set default energy set. ">vrna_md_defaults_energy_set()</a> </td></tr>
<tr>
<td>backtrack </td><td><a class="el" href="group__model__details.html#ga90da1156e6883ddd68527c2830706648" title="Get default behavior for whether to backtrack secondary structures. ">vrna_md_defaults_backtrack_get()</a> </td><td><a class="el" href="group__model__details.html#ga978c468b2fe96a70d5191e3dd17d5599" title="Set default behavior for whether to backtrack secondary structures. ">vrna_md_defaults_backtrack()</a> </td></tr>
<tr>
<td>backtrack_type </td><td><a class="el" href="group__model__details.html#ga1425b4ebd0e034dead66d79becd64143" title="Get default backtrack type, i.e. which DP matrix is used. ">vrna_md_defaults_backtrack_type_get()</a> </td><td><a class="el" href="group__model__details.html#ga68305274de96b56b7799575e222560d8" title="Set default backtrack type, i.e. which DP matrix is used. ">vrna_md_defaults_backtrack_type()</a> </td></tr>
<tr>
<td>do_backtrack </td><td>this is an alias of <em>compute_bpp</em> </td><td></td></tr>
<tr>
<td>compute_bpp </td><td><a class="el" href="group__model__details.html#gaa3a537e61fbe0518673bf9f73fd820f3" title="Get the default behavior for whether to compute base pair probabilities after partition function comp...">vrna_md_defaults_compute_bpp_get()</a> </td><td><a class="el" href="group__model__details.html#gaf1b5db10f1f476767f9a95f8a78e3132" title="Set the default behavior for whether to compute base pair probabilities after partition function comp...">vrna_md_defaults_compute_bpp()</a> </td></tr>
<tr>
<td>max_bp_span </td><td><a class="el" href="group__model__details.html#gaa60f989e062fecd4d4bac89c1883da85" title="Get default maximal base pair span. ">vrna_md_defaults_max_bp_span_get()</a> </td><td><a class="el" href="group__model__details.html#ga4c4bc962f09b4480cb8499f1cf8ae4ec" title="Set default maximal base pair span. ">vrna_md_defaults_max_bp_span()</a> </td></tr>
<tr>
<td>min_loop_size </td><td><a class="el" href="group__model__details.html#ga5cc691174a75c652807dc361b617632a" title="Get default minimal loop size. ">vrna_md_defaults_min_loop_size_get()</a> </td><td><a class="el" href="group__model__details.html#gac152f1e78c1058a10261022c8dfda0f7" title="Set default minimal loop size. ">vrna_md_defaults_min_loop_size()</a> </td></tr>
<tr>
<td>window_size </td><td><a class="el" href="group__model__details.html#ga670146a9aa3ba77f4d422d60b7c30ac9" title="Get default window size for sliding window structure prediction approaches. ">vrna_md_defaults_window_size_get()</a> </td><td><a class="el" href="group__model__details.html#ga7b802ce0e8c3181bf5cb580de6d5b26a" title="Set default window size for sliding window structure prediction approaches. ">vrna_md_defaults_window_size()</a> </td></tr>
<tr>
<td>oldAliEn </td><td><a class="el" href="group__model__details.html#ga2374492b5019df88022fe4c05f0f3630" title="Get default behavior for whether to use old energy model for comparative structure prediction...">vrna_md_defaults_oldAliEn_get()</a> </td><td><a class="el" href="group__model__details.html#ga41521d5b9fb7e0f31e7ea73f5792afab" title="Set default behavior for whether to use old energy model for comparative structure prediction...">vrna_md_defaults_oldAliEn()</a> </td></tr>
<tr>
<td>ribo </td><td><a class="el" href="group__model__details.html#ga169027f0c0561ea7d87b655e4b336bfc" title="Get default behavior for whether to use Ribosum Scoring in comparative structure prediction. ">vrna_md_defaults_ribo_get()</a> </td><td><a class="el" href="group__model__details.html#ga937c45e1d06fd6168730a9b08d130be3" title="Set default behavior for whether to use Ribosum Scoring in comparative structure prediction. ">vrna_md_defaults_ribo()</a> </td></tr>
<tr>
<td>cv_fact </td><td><a class="el" href="group__model__details.html#gae59c68393807217b0a2497adb64d3ee3" title="Get the default co-variance scaling factor used in comparative structure prediction. ">vrna_md_defaults_cv_fact_get()</a> </td><td><a class="el" href="group__model__details.html#gad3a3f40baafd91a6ce80a91a68e20053" title="Set the default co-variance scaling factor used in comparative structure prediction. ">vrna_md_defaults_cv_fact()</a> </td></tr>
<tr>
<td>nc_fact </td><td><a class="el" href="group__model__details.html#ga7ac759eaa7159bf5f022745f5da59508">vrna_md_defaults_nc_fact_get()</a> </td><td><a class="el" href="group__model__details.html#gac35e596c850dce3ad55c49119fd7d471">vrna_md_defaults_nc_fact()</a> </td></tr>
<tr>
<td>sfact </td><td><a class="el" href="group__model__details.html#gab2df6aab954b63fd3592d18e90285dae" title="Get the default scaling factor used to avoid under-/overflows in partition function computation...">vrna_md_defaults_sfact_get()</a> </td><td><a class="el" href="group__model__details.html#ga3f73d3029d3d0025d4cc311510cd95a3" title="Set the default scaling factor used to avoid under-/overflows in partition function computation...">vrna_md_defaults_sfact()</a> </td></tr>
</table>
</dd>
<dt><a class="anchor" id="_wrappers000056"></a>Global <a class="el" href="group__pf__fold.html#gaa6b8983b559b9ef4b2e1b31113ea317b">vrna_mean_bp_distance</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b><a class="el" href="group__pf__fold.html#ga79cbc375af65f11609feb6b055269e7d" title="Get the mean base pair distance of the last partition function computation. ">mean_bp_distance()</a></b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000046"></a>Global <a class="el" href="group__mfe__fold.html#gabd3b147371ccf25c577f88bbbaf159fd">vrna_mfe</a>  (vrna_fold_compound_t *vc, char *structure)</dt>
<dd>This function is attached as method <b>mfe()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000047"></a>Global <a class="el" href="group__mfe__cofold.html#gaab22d10c1190f205f16a77cab9d5d3ee">vrna_mfe_dimer</a>  (vrna_fold_compound_t *vc, char *structure)</dt>
<dd>This function is attached as method <b>mfe_dimer()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000048"></a>Global <a class="el" href="group__local__mfe__fold.html#ga689df235a1915a1ad56e377383c044ce">vrna_mfe_window</a>  (vrna_fold_compound_t *vc, FILE *file)</dt>
<dd>This function is attached as method <b>mfe_window()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000052"></a>Global <a class="el" href="group__energy__parameters.html#gac40dc82e48a72a97cfc58b9da08a7792">vrna_params_reset</a>  (vrna_fold_compound_t *vc, vrna_md_t *md_p)</dt>
<dd><p class="startdd">This function is attached to <a class="el" href="group__fold__compound.html#structvrna__fc__s" title="The most basic data structure required by many functions throughout the RNAlib. ">vrna_fc_s</a> objects as overloaded <b>params_reset()</b> method.</p>
<p class="enddd">When no parameter is passed to this method, the resulting action is the same as passing <em>NULL</em> as second parameter to <a class="el" href="group__energy__parameters.html#gac40dc82e48a72a97cfc58b9da08a7792" title="Reset free energy parameters within a vrna_fold_compound_t according to provided, or default model de...">vrna_params_reset()</a>, i.e. global default model settings are used. Passing an object of type <a class="el" href="group__model__details.html#structvrna__md__s" title="The data structure that contains the complete model details used throughout the calculations. ">vrna_md_s</a> resets the fold compound according to the specifications stored within the <a class="el" href="group__model__details.html#structvrna__md__s" title="The data structure that contains the complete model details used throughout the calculations. ">vrna_md_s</a> object.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000050"></a>Global <a class="el" href="group__energy__parameters.html#ga5d1909208f7ea3baa98b75afaa1f62ca">vrna_params_subst</a>  (vrna_fold_compound_t *vc, vrna_param_t *par)</dt>
<dd>This function is attached to <a class="el" href="group__fold__compound.html#structvrna__fc__s" title="The most basic data structure required by many functions throughout the RNAlib. ">vrna_fc_s</a> objects as <b>params_subst()</b> method.  </dd>
<dt><a class="anchor" id="_wrappers000058"></a>Global <a class="el" href="group__subopt__stochbt.html#ga901fe42a33b07be083421741bf7dc610">vrna_pbacktrack</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as overloaded method <b><a class="el" href="group__subopt__stochbt.html#gac03ca6db186bb3bf0a2a326d7fb3ba03" title="Sample a secondary structure from the Boltzmann ensemble according its probability. ">pbacktrack()</a></b> to objects of type <em>fold_compound</em> that accepts an optional <code>length</code> argument. Hence, it serves as a replacement for <a class="el" href="group__subopt__stochbt.html#ga901fe42a33b07be083421741bf7dc610" title="Sample a secondary structure (consensus structure) from the Boltzmann ensemble according its probabil...">vrna_pbacktrack()</a>.  </dd>
<dt><a class="anchor" id="_wrappers000057"></a>Global <a class="el" href="group__subopt__stochbt.html#ga347375a4da8bdff74639529847f97d4f">vrna_pbacktrack5</a>  (vrna_fold_compound_t *vc, int length)</dt>
<dd>This function is attached as overloaded method <b><a class="el" href="group__subopt__stochbt.html#gac03ca6db186bb3bf0a2a326d7fb3ba03" title="Sample a secondary structure from the Boltzmann ensemble according its probability. ">pbacktrack()</a></b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000054"></a>Global <a class="el" href="group__pf__fold.html#ga29e256d688ad221b78d37f427e0e99bc">vrna_pf</a>  (vrna_fold_compound_t *vc, char *structure)</dt>
<dd>This function is attached as method <b>pf()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000055"></a>Global <a class="el" href="group__pf__cofold.html#ga4e5c7d06c302a7c59fc0d64dc142ca63">vrna_pf_dimer</a>  (vrna_fold_compound_t *vc, char *structure)</dt>
<dd>This function is attached as method <b>pf_dimer()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000018"></a>Global <a class="el" href="group__soft__constraints.html#gaf162aedac7422f2eb16ea030f47d2f4b">vrna_sc_add_bp</a>  (vrna_fold_compound_t *vc, int i, int j, FLT_OR_DBL energy, unsigned int options)</dt>
<dd><p class="startdd"></p>
<p class="enddd">This function is attached as an overloaded method <b>sc_add_bp()</b> to objects of type <em>fold_compound</em>. The method either takes arguments for a single base pair (i,j) with the corresponding energy value: </p><div class="fragment"><div class="line">fold_compound.sc_add_bp(i, j, energy, options)</div></div><!-- fragment --><p> or an entire 2-dimensional matrix with dimensions n x n that stores free energy contributions for any base pair (i,j) with <img class="formulaInl" alt="$ 1 \leq i &lt; j \leq n $" src="form_123.png"/>: </p><div class="fragment"><div class="line">fold_compound.sc_add_bp(matrix, options)</div></div><!-- fragment --><p> In both variants, the <code>options</code> argument is optional can may be omitted.  </p>
</dd>
<dt><a class="anchor" id="_wrappers000023"></a>Global <a class="el" href="group__soft__constraints.html#gabde7d07a79bb9a8f4721aee247b674ea">vrna_sc_add_bt</a>  (vrna_fold_compound_t *vc, vrna_callback_sc_backtrack *f)</dt>
<dd>This function is attached as method <b>sc_add_bt()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000021"></a>Global <a class="el" href="group__soft__constraints.html#ga15c6d52471ec97897e2bb7f964f5deb6">vrna_sc_add_data</a>  (vrna_fold_compound_t *vc, void *data, vrna_callback_free_auxdata *free_data)</dt>
<dd>This function is attached as method <b>sc_add_data()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000024"></a>Global <a class="el" href="group__soft__constraints.html#ga87e382b5d0c9b7d9ce1b79c0473ff700">vrna_sc_add_exp_f</a>  (vrna_fold_compound_t *vc, vrna_callback_sc_exp_energy *exp_f)</dt>
<dd>This function is attached as method <b>sc_add_exp_f()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000022"></a>Global <a class="el" href="group__soft__constraints.html#ga8c7d907ec0125cd61c04e0908010a4e9">vrna_sc_add_f</a>  (vrna_fold_compound_t *vc, vrna_callback_sc_energy *f)</dt>
<dd>This function is attached as method <b>sc_add_f()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000011"></a>Global <a class="el" href="group__constraints__ligand.html#gaa6ff0113a3a76dc0b8d62961f4e1dfa0">vrna_sc_add_hi_motif</a>  (vrna_fold_compound_t *vc, const char *seq, const char *structure, FLT_OR_DBL energy, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_add_hi_motif()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000012"></a>Global <a class="el" href="group__SHAPE__reactivities.html#ga57d612b58e1c61dd6cfcb5a843f8f1b3">vrna_sc_add_SHAPE_deigan</a>  (vrna_fold_compound_t *vc, const double *reactivities, double m, double b, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_add_SHAPE_deigan()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000013"></a>Global <a class="el" href="group__SHAPE__reactivities.html#ga04ba85da63d8c793bb8001d1e6f800ba">vrna_sc_add_SHAPE_deigan_ali</a>  (vrna_fold_compound_t *vc, const char **shape_files, const int *shape_file_association, double m, double b, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_add_SHAPE_deigan_ali()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000014"></a>Global <a class="el" href="group__SHAPE__reactivities.html#gaf3c65a045060aef5c4e41693d30af58c">vrna_sc_add_SHAPE_zarringhalam</a>  (vrna_fold_compound_t *vc, const double *reactivities, double b, double default_value, const char *shape_conversion, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_add_SHAPE_zarringhalam()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000020"></a>Global <a class="el" href="group__soft__constraints.html#ga069915fe203a2c8e522dd37847177a09">vrna_sc_add_up</a>  (vrna_fold_compound_t *vc, int i, FLT_OR_DBL energy, unsigned int options)</dt>
<dd>This function is attached as an overloaded method <b>sc_add_up()</b> to objects of type <em>fold_compound</em>. The method either takes arguments for a single nucleotide <img class="formulaInl" alt="$i $" src="form_124.png"/> with the corresponding energy value: <div class="fragment"><div class="line">fold_compound.sc_add_up(i, energy, options)</div></div><!-- fragment --> or an entire vector that stores free energy contributions for each nucleotide <img class="formulaInl" alt="$i $" src="form_124.png"/> with <img class="formulaInl" alt="$ 1 \leq i \leq n $" src="form_125.png"/>: <div class="fragment"><div class="line">fold_compound.sc_add_bp(vector, options)</div></div><!-- fragment --> In both variants, the <code>options</code> argument is optional can may be omitted.  </dd>
<dt><a class="anchor" id="_wrappers000015"></a>Global <a class="el" href="group__soft__constraints.html#ga9d977a1681356778cc66dceafbe5b032">vrna_sc_init</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b>sc_init()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000016"></a>Global <a class="el" href="group__soft__constraints.html#ga73cdc07b9a199c614367bebef0f2c41a">vrna_sc_remove</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b>sc_remove()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000017"></a>Global <a class="el" href="group__soft__constraints.html#ga8e4334b24bc91453fbcda490a4e331af">vrna_sc_set_bp</a>  (vrna_fold_compound_t *vc, const FLT_OR_DBL **constraints, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_set_bp()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000019"></a>Global <a class="el" href="group__soft__constraints.html#ga99ed63f3ef9e7fe3997932030487a344">vrna_sc_set_up</a>  (vrna_fold_compound_t *vc, const FLT_OR_DBL *constraints, unsigned int options)</dt>
<dd>This function is attached as method <b>sc_set_up()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000061"></a>Global <a class="el" href="group__subopt__wuchty.html#ga0f11d738fb8c8b1885a90c11c8931ff6">vrna_subopt</a>  (vrna_fold_compound_t *vc, int delta, int sorted, FILE *fp)</dt>
<dd>This function is attached as method <b><a class="el" href="group__subopt__wuchty.html#ga700f662506a233e42dd7fda74fafd40e" title="Returns list of subopt structures or writes to fp. ">subopt()</a></b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000062"></a>Global <a class="el" href="group__subopt__wuchty.html#ga1053837e6b6f158093508f8a70998352">vrna_subopt_cb</a>  (vrna_fold_compound_t *vc, int delta, vrna_subopt_callback *cb, void *data)</dt>
<dd>This function is attached as method <b>subopt_cb()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000063"></a>Global <a class="el" href="group__subopt__zuker.html#gababde9d210eb433854f1e71da7815fbb">vrna_subopt_zuker</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b>subopt_zuker()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000040"></a>Global <a class="el" href="group__domains__up.html#gaec0c3313fb2951946614f920d289829a">vrna_ud_add_motif</a>  (vrna_fold_compound_t *vc, const char *motif, double motif_en, unsigned int loop_type)</dt>
<dd>This function is attached as method <b>ud_add_motif()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000041"></a>Global <a class="el" href="group__domains__up.html#gada59cb0c498b812eadd010811af3f2d4">vrna_ud_remove</a>  (vrna_fold_compound_t *vc)</dt>
<dd>This function is attached as method <b>ud_remove()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000042"></a>Global <a class="el" href="group__domains__up.html#gac1f18c312b91d80089534a87d956e58b">vrna_ud_set_data</a>  (vrna_fold_compound_t *vc, void *data, vrna_callback_free_auxdata *free_cb)</dt>
<dd>This function is attached as method <b>ud_set_data()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000044"></a>Global <a class="el" href="group__domains__up.html#ga2fb1db2099da26c76247e1209ad4aa09">vrna_ud_set_exp_prod_rule_cb</a>  (vrna_fold_compound_t *vc, vrna_callback_ud_exp_production *pre_cb, vrna_callback_ud_exp_energy *exp_e_cb)</dt>
<dd>This function is attached as method <b>ud_set_exp_prod_rule_cb()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000045"></a>Global <a class="el" href="unstructured__domains_8h.html#a13ac877c9db89a1a5b5d9c0394148595">vrna_ud_set_prob_cb</a>  (vrna_fold_compound_t *vc, vrna_callback_ud_probs_add *setter, vrna_callback_ud_probs_get *getter)</dt>
<dd>This function is attached as method <b>ud_set_prob_cb()</b> to objects of type <em>fold_compound</em>  </dd>
<dt><a class="anchor" id="_wrappers000043"></a>Global <a class="el" href="group__domains__up.html#ga745a99f0bc72898d54de16f6e538828a">vrna_ud_set_prod_rule_cb</a>  (vrna_fold_compound_t *vc, vrna_callback_ud_production *pre_cb, vrna_callback_ud_energy *e_cb)</dt>
<dd>This function is attached as method <b>ud_set_prod_rule_cb()</b> to objects of type <em>fold_compound</em> </dd>
</dl>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Tue Nov 14 2017 10:01:20 for RNAlib-2.4.3 by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.13 </li>
  </ul>
</div>
</body>
</html>
